sources/scrapy_scraper/spider.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Dict, Generator, List, Optional
+
+import scrapy  # type: ignore
+from scrapy.responsetypes import Response  # type: ignore
+
+from .types import BaseQueue
+
+
+class DLTSpiderBase(scrapy.Spider):
+    """Base spider which requires additional
+    dependencies and subsequent spiders should
+    implement their own parse logic with sending
+    the extracted data over the queue.
+    """
+
+    def __init__(
+        self,
+        queue: BaseQueue,
+        name: Optional[str] = None,
+        settings: Optional[Dict[str, Any]] = None,
+        start_urls: Optional[List[str]] = None,
+        **kwargs: Any,
+    ):
+        super().__init__(name, **kwargs)
+        self._queue = queue
+        self.custom_settings = settings or {}
+        self.start_urls = start_urls or []
+
+    def send_data(self, data: Any) -> None:
+        self._queue.put(data)
+
+    def done(self) -> None:
+        self.send_data({"done": True})
+
+
+"""
+Below is a slightly modified example of Scrapy spider
+which you find in the official documentation
+https://docs.scrapy.org/en/latest/intro/tutorial.html#extracting-data-in-our-spider.
+
+To send the data from the spider to pipeline the spider needs a queue instance,
+once scrapy closes the session we also send a termination flag so
+DLT pipeline can stop waiting for more data.
+"""
+
+
+class QuotesSpider(DLTSpiderBase):
```

Comment:
Since this is example it's best to move it out of the common code and into the example pipeline file `scrapy_scraper_pipeline.py`.